pymatgen/entries/computed_entries.py

```diff
@@ -86,21 +271,45 @@ def normalize(self, mode: str = "formula_unit") -> None:
                 normalizes such that the composition amounts sum to 1.
         """
         factor = self._normalization_factor(mode)
-        self.correction /= factor
         self.uncorrected_energy /= factor
+        for e in self.energy_adjustments:
```

Comment:
Is this new? It seems like an odd function.

Reply:
Yes it is new. As you noted above, whenever we scale the composition (as `normalize` does), we need to rescale all the `EnergyAdjustment` too. This is why I had added the setters. I'm having trouble figuring out how to make a combination of regular attrs and @property work here.

In order to make `value` and abstract method, it has to be a @property, right? In that case, I have to have setters for every `EnergyAdjustment` class to enable scaling. I've tried with a mixture of @property and regular attributes for the different classes but can't seem to make it work.

Reply:
I would put this into the energy adjustment as an abstract method then. But in general, I don't think it's necessary. Any adjustment should only be valid within the context of the composition it applies to. If we want an adjustment for a similar composition, we should go through compatibility.